### ISSUE TITLE
Export & Import as BREP and STEP

### ISSUE THREAD

**Issue by 5263** [bjnortier/shapesmith.deprecated#65]
Would be useful for thingiverse integration.

**Comment by bjnortier** [comment]
STEP yes.
Do you mean OpenCASCADE BREP format? Does FreeCAD use this?

**Comment by 5263** [comment]
Yes and yes. pythonocc, Cadmium and openshapefactory allow export in OpenCASCADE Brep File format as well.

**Comment by bjnortier** [comment]
Ok, that shouldn't be too hard. Assuming it's the format already used to serialize brep

**Comment by 5263** [comment]
The limitation is basicly that a brep file contains only one shape. So maybe one has to build a Compound first.

```
# include <BRepTools.hxx>
BRepTools::Write (const TopoDS_Shape& sh, const Standard_CString file)
```

STEP (or IGES) would be more complicated, depending how much information you want to support and if you use OCAF.

**Comment by bjnortier** [comment]
Thanks, yes. That's what is used to serialise to disk.
Shapesmith doesn't use OCAF.
